### DISCUSSION TITLE
Crack propagation

### DISCUSSION THREAD

**Discussion by Nikhil23121998** [peridigm/peridigm#294] [Q&A] [answered]
Which sample can I choose for crack propagation analysis in peridigm, and what kind of modification needs to be done in that sample?

**Comment by oldninja** [comment, marked as answer]
You can start here to understand the syntax and options: https://github.com/peridigm/peridigm/tree/master/examples%2Fdisk_impact

But you may want to create your own test cases. You can also look into the tests folders for simpler examples.

**Reply by Nikhil23121998** [reply]
Thankyou

**Reply by Nikhil23121998** [reply]
I was working on the pre-existing crack test sample. To see crack propagation, do I have to modify the .g file and give crack on the test sample?

**Reply by oldninja** [reply]
You will need to give a discretization bond filter on the *.yaml file:
`  Discretization:  `
`      Type: "Text File"  `
`      Input Mesh File: "model.g"  `
`      Bond Filters:  `
`        Precrack Bond Filter:  `
`          Type: "Rectangular_Plane"  `
`          Normal_X: value  `
`          Normal_Y: value  `
`          Normal_Z: value  `
`          Lower_Left_Corner_X: value  `
`          Lower_Left_Corner_Y: value  `
`          Lower_Left_Corner_Z: value  `
`          Bottom_Unit_Vector_X: value  `
`          Bottom_Unit_Vector_Y: value  `
`          Bottom_Unit_Vector_Z: value  `
`          Bottom_Length: value  `
`          Side_Length: value  `

**Reply by Nikhil23121998** [reply]
While I am running my yaml file, this error is showing,

-- Peridigm
-- version 1.5.0 (Dev)

error: Parser failure at line 27 column 2 of Plate.yaml
Blocks:
^

/home/nikhil/trilinos/packages/teuchos/parameterlist/src/Teuchos_YamlParser.cpp:252:

Throw number = 1

Throw test that evaluated to true: !rhs.at(0).empty()

Can't specify multiple top-level ParameterLists in one YAML file!

Wallclock Time (seconds):
  Total             0.19

**Reply by oldninja** [reply]
Looks like it may be just an indentation issue or something like that.

**Reply by Nikhil23121998** [reply]
This is the YAML file I am using. 

Verbose: false
Discretization: 
     Type: "Exodus" 
     Input Mesh File: "Plate.g" 
     Bond Filters: 
       Precrack Bond Filter: 
         Type: "Rectangular_Plane" 
         Normal_X: 0.0 
         Normal_Y: 1.0 
         Normal_Z: 0.0 
         Lower_Left_Corner_X: -10.0 
         Lower_Left_Corner_Y: 0.0 
         Lower_Left_Corner_Z: -10.0 
         Bottom_Unit_Vector_X: 1.0 
         Bottom_Unit_Vector_Y: 0.0 
         Bottom_Unit_Vector_Z: 0.0 
         Bottom_Length: 10 
         Side_Length: 20 

Materials:
      My Material:
          Material Model: "Elastic"
          Density: 8.0
          Bulk Modulus: 1.500e12
          Shear Modulus: 6.923e11

Blocks:
     My Block:
          Block Names: block_1
          Material: My Material
          Horizon: 3.015

Boundary Conditions:
      Prescribed Displacement Bottom:
        Type: "Prescribed Displacement"
        Node Set: "nodelist_1"
        Coordinate: "y"
        Value: "y*0.01*t"
      Prescribed Displacement Top:
        Type: "Prescribed Displacement"
        Node Set: "nodelist_2"
        Coordinate: "y"
        Value: "y*0.01*t"
      Prescribed Displacement Fix Bottom Rigid Body Motion In Z:
        Type: "Prescribed Displacement"
        Node Set: "nodelist_3"
        Coordinate: "z"
        Value: "0.0"
      Prescribed Displacement Fix Bottom Rigid Body Motion In X:
        Type: "Prescribed Displacement"
        Node Set: "nodelist_4"
        Coordinate: "x"
        Value: "0.0"

Solver:
      Verbose: false
      Initial Time: 0.0
      Final Time: 1.0
      QuasiStatic
        Number of Load Steps: 1
        Absolute Tolerance: 1.0
        Maximum Solver Iterations: 10
Output:
     Output File Type: ExodusII
     Output Format: BINARY
     Output Filename: PrecrackedPlate
     Output Frequency: 1
     Parallel Write: true

Output Variables:
     Displacement: true
     Velocity: true
     Element_Id: true
     Proc_Num: true
     Dilatation: true
     Force_Density: true
     Force: true
     Volume: true
     Number_Of_Neighbors: true

**Reply by Nikhil23121998** [reply]
When I am running XML file, my output is coming like that-

![Screenshot (59)](https://user-images.githubusercontent.com/110670051/213928077-7e0e5ef6-e871-43f1-ae05-aab0a298895a.png)

**Reply by oldninja** [reply]
It is working then because you can see the reduced number of neighbors in the contour. Maybe check the format of your yaml file... It may be just an indentation or a character issue somewhere in the deck. Can you share the actual yaml file here?

**Reply by Nikhil23121998** [reply]
This is my YAML file 

![Screenshot (60)](https://user-images.githubusercontent.com/110670051/213989998-fb904c85-1c24-4d69-b949-f47330b87457.png)
![Screenshot (61)](https://user-images.githubusercontent.com/110670051/213990048-c94cd60a-3726-45ef-8efb-12ef20c901f2.png)
![Screenshot (62)](https://user-images.githubusercontent.com/110670051/213990083-5b2b1287-5cc1-431f-8539-27fc486604c3.png)

**Reply by Nikhil23121998** [reply]
After some changes my YAML file is now working, but this is how my output file looking, 

~/peridigm/test/regression/PrecrackedPlate$ /home/nikhil/peridigm/bin/Peridigm Plate.yaml

-- Peridigm
-- version 1.5.0 (Dev)

Allocating global tangent matrix...
  number of rows = 1920
  number of nonzeros = 1339236

Load step 1, initial time = 0, final time = 0.25, convergence criterion = 1
  iteration 1: residual = 4.95214e+11
  iteration 2: residual = 2.30951e+09
  iteration 3: residual = 132767
  iteration 4: residual = 13.3868
  iteration 5: residual = 0.0371762
  cpu time for load step = 2.7 sec., cumulative cpu time = 2.7 sec.

Load step 2, initial time = 0.25, final time = 0.5, convergence criterion = 1
  iteration 1: residual = 5e+11
  iteration 2: residual = 4.9e+08
  iteration 3: residual = 5.8e+04
  iteration 4: residual = 4.8
  iteration 5: residual = 0.039
  cpu time for load step = 1.8 sec., cumulative cpu time = 4.5 sec.

Load step 3, initial time = 0.5, final time = 0.75, convergence criterion = 1
  iteration 1: residual = 5e+11
  iteration 2: residual = 4.9e+08
  iteration 3: residual = 5.8e+04
  iteration 4: residual = 4.9
  iteration 5: residual = 0.035
  cpu time for load step = 1.9 sec., cumulative cpu time = 6.4 sec.

Load step 4, initial time = 0.75, final time = 1, convergence criterion = 1
  iteration 1: residual = 5e+11
  iteration 2: residual = 4.9e+08
  iteration 3: residual = 5.7e+04
  iteration 4: residual = 5.4
  iteration 5: residual = 0.035
  cpu time for load step = 2.1 sec., cumulative cpu time = 8.5 sec.


Memory Usage (Heap Alloc MB):
  Allocated Jacobian                      35
  Post Execute                            22
  Zoltan Search Tree                     3.9

Wallclock Time (seconds):
  Total                                  9.7
  Solve Linear System                    1.9
  Output                               0.053
  Line Search                           0.19
  Internal Force                        0.21
  Gather/Scatter                      0.0067
  Evaluate Jacobian                      5.6
  Compute Residual                      0.23
  Apply Kinematic B.C.               0.00095
  Apply Initial Conditions           3.4e-06
  Apply Boundary Conditions             0.78
  Apply Body Forces                  1.2e-05
  Allocate Global Tangent                  1
![Screenshot (63)](https://user-images.githubusercontent.com/110670051/214121362-be65ea2a-a182-4399-a87c-8d1e833c798b.png)
[New Text Document.txt](https://github.com/peridigm/peridigm/files/10482677/New.Text.Document.txt)

**Comment by andrisfreimanis** [comment]
Yaml files should start with a top level parameter list called "Peridigm:".

Something like in the picture:
[cid:b0a86d96-637b-4fbf-b031-e00a31f1c9b7]

Best regards
Andris
________________________________

**Reply by Nikhil23121998** [reply]
Still showing the same error.

**Comment by andrisfreimanis** [comment]
Also check that you do not have any tab characters. Indent only with spaces.

This error might be related to that.
"error: Parser failure at line 27 column 2 of Plate.yaml
Blocks:
^"

Andris


________________________________

**Reply by Nikhil23121998** [reply]
[New Text Document.txt](https://github.com/peridigm/peridigm/files/10477893/New.Text.Document.txt)

This is my YAML file
